### DISCUSSION TITLE
Proposed collaboration with freemathapp.org, more flexible free response answer collection

### DISCUSSION THREAD

**Discussion by jaltekruse** [openwebwork/pg#671] [Ideas]
Hello WebWork team,

My name is Jason Altekruse and I am the founder of https://freemathapp.org. The site's main focus is enabling students to show math work digitally and give a quick grading experience for instructors. The student experience includes the ability to quickly edit a series of Mathquills or text boxes, along with the ability to upload images or draw on a digital canvas or grid.

The site currently has no back-end and relies on LMSes to transfer files full of student work to teachers for manual grading.

I know lots of institutions like to have at least partly automated grading for assessments, so I'm reaching out to platforms like WebWork to see if you might be interested in embedding the experience from Free Math to give students a more flexible workspace for free response type questions. I have seen that you have an essay question type, and I'm not sure how much it might have changed in version 3. Is it still mostly focused on letting students type a combination of text and latex like in this screenshot? Is there any way for students to draw a graphical response or include an image?

I do know this screenshot is from an old blog, so things might have evolved significantly since this was taken. I'll be downloading the repo and trying to get a local instance running so I can test out the current experience on WebWork.
![image](https://user-images.githubusercontent.com/365366/163505045-588bf788-b9dc-43b3-81af-f5cbef27c1ec.png)

**Comment by jaltekruse** [comment]
Note this was originally posted on the webwork3 repo, but I reposted here based on a recommendation by @dlglin.

https://github.com/openwebwork/webwork3/issues/84

**Comment by jaltekruse** [comment]
Any thoughts on this @pstaabp, @drgrice1, @mgage, @taniwallach, @drdrew42 ?

**Comment by drgrice1** [comment]
The essay type questions have not changed.  There is not a built in way for students to draw or include images or files.  Although there have been feature requests for the ability to directly attach files to a problem.  That is something that will come sooner or later.

**Reply by jaltekruse** [reply]
Thanks for the reply @drgrice1!

If there is interest in an interface for students that would allow these types of inputs, graphical math expression editing, easily showing step-by-step reasoning and the ability to embed drawing or images I would be happy to spend some time building this into WebWork myself and open a PR. I just wanted to confirm if there would potentially be interest from the community/maintainers before I put in significant effort.
